var/spack/repos/builtin/packages/aluminum/package.py

```diff
@@ -62,8 +67,13 @@ class Aluminum(CMakePackage, CudaPackage, ROCmPackage):
     depends_on("cub", when="@:0.1,0.6.0: +cuda ^cuda@:10")
     depends_on("hipcub", when="@:0.1,0.6.0: +rocm")
 
+    depends_on("rccl", when="+rccl")
+    depends_on("aws-ofi-rccl", when="+libfabric platform=cray")
+
     conflicts("~cuda", when="+cuda_rma", msg="CUDA RMA support requires CUDA")
     conflicts("+cuda", when="+rocm", msg="CUDA and ROCm support are mutually exclusive")
+    conflicts("+nccl", when="+rccl", msg="NCCL and RCCL support are mutually exclusive")
+    conflicts("~rccl", when="+libfabric", msg="libfabric enhancements require RCCL support")
```

Comment:
Not familiar with aluminum, but I'm curious why this is the case. This in itself does not guarantee that libfabric's ROCm support is enabled (so p2p transfers with FI_HMEM_ROCR capability might not be supported).  From a middleware's perspective, you might be better off querying HMEM capabilities you need at fi_getinfo() to be sure. 

This is pointing out a gap we have in the libfabric package.py though. We need to add the CUDA/ROCR build variants that also pull in the respective runtimes as deps (like how you've done for the aws-ofi-rccl package).

Reply:
@rajachan Hmmm, I am not too sure of what you are suggesting.  I guess I put this in here now because we don't have any software combination that would allow us to use the `rccl` `libfabric` plugin without already having `rccl` support. Perhaps this variant should be renamed because it is not tying `libfabric` support directly into `aluminum`, it is just adding the dependency on the `aws-ofi-rccl` plugin.

Reply:
Yeah, the name is what threw me off. Looks good now.